Autopilot/PathManager/Src/landingTakeoffManager.cpp

```diff
@@ -0,0 +1,184 @@
+#include "waypointManager.hpp"
+#include "landingTakeoffManager.hpp"
+#include <math.h>
+#include "vectorClass.hpp"
+
+double LandingManager::changingAltitude(_WaypointManager_Data_In input, _PathData aimingPoint, _PathData intersectionPoint, _PathData stoppingPoint)
+{
+    //vector declaration
+    Vector3D aPoint(aimingPoint.longitude, aimingPoint.latitude, aimingPoint.altitude); //aiming point
+    Vector3D iPoint(intersectionPoint.longitude, intersectionPoint.latitude, intersectionPoint.altitude); //intersection point
+    Vector3D sPoint(stoppingPoint.longitude, stoppingPoint.latitude, stoppingPoint.altitude); //stopping point
+    Vector3D cPoint(input.longitude, input.latitude, input.altitude); //current point
+    Vector3D projectedPoint;
+
+    //calculating the vectors
+    Vector3D vectorAI(iPoint.x - aPoint.x, iPoint.y - aPoint.y, iPoint.z - aPoint.z);
+    Vector3D vectorAS(sPoint.x - aPoint.x, sPoint.y - aPoint.y, sPoint.z - aPoint.z);
+    Vector3D vectorAC(cPoint.x - aPoint.x, cPoint.y - aPoint.y, cPoint.z - aPoint.z);
+    Vector3D normal;
+    
+    //normal of the plane created by aiming, stopping, and intersection point
+    normal = vectorAI.crossProduct(vectorAS);
+
+    //projecting vectorAC to the normal
+    //equation is the dot product of normal and vectorAC, divided by the norm of the normal squared, multiplied by the normal vector
+    Vector3D projectionACToNormal;
+
+    projectionACToNormal = normal * (normal.dotProduct(vectorAC) / pow(normal.norm(),2));
+
+    //project point onto plane (cpoint minus the projection)
+
+    projectedPoint = cPoint - projectionACToNormal;
+
+    //determine the equation of the line of the slope and sub in projected values (symmetric form)
+    double altitude = (((projectedPoint.x - aPoint.x) / vectorAI.x) + aPoint.z) * vectorAI.z;
+
+    return altitude;
+}
+
+double LandingManager::throttleOff(void)
+{
+    //return 0 for now, this function needs to be fixed
+    return 0;
+}
+
+double LandingManager::approachSpeed(double windSpeed, bool ifPackage)
+{
+    //approach speed calculation for both package and no package scenarios
+    if(ifPackage)
+    {
+        return STALL_SPEED_WITH_PACKAGE * 1.3 + windSpeed;
+    }
+    else 
+    {
+        return STALL_SPEED_NO_PACKAGE * 1.3 + windSpeed;
+    }
+}
+
+double LandingManager::slowFlightSpeed(bool ifPackage)
+{
+    //slow flight speed calculation for both package and no package scenarios
+    if(ifPackage)
+    {
+        return STALL_SPEED_WITH_PACKAGE + 2;
+    }
+    else 
+    {
+        return STALL_SPEED_NO_PACKAGE + 2;
+    }
```

Comment:
would it make sense to make it slightly faster (to account for some more error)?

Reply:
these values will need to be tuned when we're simulating the landing takeoff stuff, they're not set in stone.